### PR TITLE
fix: correct the totalCount value returned by some list API

### LINKED ISSUE #484: The interface returned an incorrect totalCount value after keyword search on the model warehouse list page
模型仓库列表页关键字搜索后接口返回的totalCount值错误

### PR DESCRIPTION
**Pull request by dayuy**
#### What type of PR is this?
bug

#### What this PR does / why we need it
Correct the totalCount value returned by some list API
#### Which issue(s) this PR fixes

Fixes #484

### CODE REVIEW

File: apiserver/pkg/application/application.go [hunk 1]
```diff
@@ -309,6 +309,12 @@ func ListApplicationMeatadatas(ctx context.Context, c dynamic.Interface, input g
 	}
 	totalCount := len(filtered)
 
+	if page < 0 {
```
Comment (0xff-dev) [inline apiserver/pkg/application/application.go:312]: `if page <= 0`

File: apiserver/pkg/application/application.go [hunk 2]
```diff
@@ -309,6 +309,12 @@ func ListApplicationMeatadatas(ctx context.Context, c dynamic.Interface, input g
...
+		page = 1
+	}
+	if pageSize < 0 {
```
Comment (0xff-dev) [inline apiserver/pkg/application/application.go:315]: A function can be created to calculate the start and end positions of the paging.
@Abirdcfly @bjwswang Any suggestions?
```go
func PagePosition(page, size, total int) (int ,int) {
    start := (page-1)*size
    end :=  page * size
    if start > total {
        start = total
    }
    if end > total {
      end = total
    }
    return start, end
}
```

Comment (bjwswang) [reply apiserver/pkg/application/application.go]: Great if we have this common func.

Comment (dayuy) [reply apiserver/pkg/application/application.go]: done

File: apiserver/pkg/common/common.go [hunk 3]
```diff
@@ -198,3 +198,26 @@ func SetCreator(ctx context.Context, common *v1alpha1.CommonSpec) {
 	currentUser, _ := ctx.Value(auth.UserNameContextKey).(string)
 	common.Creator = currentUser
 }
+
+func PagePosition(page, size, total int) (int, int) {
```
Comment (0xff-dev) [inline apiserver/pkg/common/common.go:202]: Good job!

**Review by 0xff-dev — APPROVED** [review]

File: apiserver/pkg/common/common.go (hunk 3 above)
Comment (bjwswang) [reply apiserver/pkg/common/common.go:202]: 👍

**Review by bjwswang — APPROVED** [review]

**Pull request merged**